e2e/app/eoa/eoa.go

```diff
@@ -4,85 +4,128 @@ package eoa
+	PrivateKey *ecdsa.PrivateKey // only for devnet (well-known type)
+
+	Chains        ChainSelector // all vs specific chains
+	MinBalance    *big.Int      // check if below this balance
+	TargetBalance *big.Int      // fund to this balance
 }
 
-var (
-	devnetKeys = map[Type]*ecdsa.PrivateKey{
-		TypeRelayer: anvil.DevPrivateKey5(),
-		TypeMonitor: anvil.DevPrivateKey6(),
+func (r Role) Verify() error {
+	if r != RoleRelayer && r != RoleMonitor {
+		return errors.New("invalid role", "role", r)
 	}
-)
 
-var secureAddrs = map[netconf.ID]map[Type]common.Address{
-	netconf.Staging: {
-		TypeRelayer: common.HexToAddress("0xfE921e06Ed0a22c035b4aCFF0A5D3a434A330c96"),
-		TypeMonitor: common.HexToAddress("0x0De553555Fa19d787Af4273B18bDB77282D618c4"),
-	},
-	netconf.Testnet: {
-		TypeRelayer: common.HexToAddress("0x01654f55E4F5E2f2ff8080702676F1984CBf7d8a"),
-		TypeMonitor: common.HexToAddress("0x12Dc870b3F5b7f810c3d1e489e32a64d4E25AaCA"),
-	},
-	netconf.Mainnet: {
-		TypeMonitor: common.HexToAddress("0x07082fcbFA5F5AC9FBc03A48B7f6391441DB8332"),
-		TypeRelayer: common.HexToAddress("0x07804D7B8be635c0C68Cdf3E946114221B12f4F7"),
-	},
+	return nil
 }
 
-// MustAddress returns the address for the EOA identified by the network and type.
-func MustAddress(network netconf.ID, typ Type) common.Address {
-	resp, _ := Address(network, typ)
+// MustAddress returns the address for the EOA identified by the network and role.
+func MustAddress(network netconf.ID, role Role) common.Address {
+	resp, _ := Address(network, role)
 	return resp
 }
 
-// Address returns the address for the EOA identified by the network and type.
-func Address(network netconf.ID, typ Type) (common.Address, bool) {
-	if network == netconf.Devnet {
-		return crypto.PubkeyToAddress(devnetKeys[typ].PublicKey), true
+// Address returns the address for the EOA identified by the network and role.
+func Address(network netconf.ID, role Role) (common.Address, bool) {
+	accounts, ok := statics[network]
+	if !ok {
+		return common.Address{}, false
 	}
 
-	resp, ok := secureAddrs[network][typ]
+	for _, account := range accounts {
+		if account.Role == role {
+			return account.Address, true
+		}
+	}
 
-	return resp, ok
+	return common.Address{}, false
 }
 
-// PrivateKey returns the private key for the EOA identified by the network and type.
-func PrivateKey(ctx context.Context, network netconf.ID, typ Type) (*ecdsa.PrivateKey, error) {
-	if network == netconf.Devnet {
-		return devnetKeys[typ], nil
-	}
-
-	addr, ok := secureAddrs[network][typ]
+// PrivateKey returns the private key for the EOA identified by the network and role.
+func PrivateKey(ctx context.Context, network netconf.ID, role Role) (*ecdsa.PrivateKey, error) {
+	acc, ok := accountForRole(network, role)
 	if !ok {
-		return nil, errors.New("eoa key not defined", "network", network, "type", typ)
+		return nil, errors.New("eoa key not defined", "network", network, "role", role)
+	}
+	if network == netconf.Devnet {
```

Comment:
this should be based on type:
- if type == wellknown, return privkey
- if type == remote, return error
- if type == secret, download